src/Symfony/Component/Security/Core/Exception/AlreadySwitchedException.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Exception;
+
+/**
+ * Thrown when trying to switch to another user while being already switched.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+final class AlreadySwitchedException extends AuthenticationException
```

Comment:
Perhaps we could extend `CustomUserMessageAuthenticationException` to provide a sensible & safe message to show to the user?

Reply:
-1, we do not do that for any other core exception. If a user wants a more detailed message, they can do so by creating a new exception in an exception listener. (which is also why this specific exception is usefull: you can catch only this one)